docs/src/modules/spring/pages/event-sourced-entities.adoc

```diff
@@ -0,0 +1,158 @@
+
+== Implementing behavior
+
+Now that we have our Entity state defined along with its events, the remaining steps can be summarized as follows:
+
+- declare your entity and pick an entity key (it needs to be unique as it will be used for sharding purposes);
+- define an access point (i.e. a route path) to your entity;
+- implement how each command is handled and which event(s) it generates;
+- provide a handler for each event and how it affects the entity's state.
+
+Let's have a look at what our shopping cart entity will look like for the first 2 steps from the above list:
+
+[source,java]
+.src/main/java/com/example/shoppingcart/ShoppingCartEntity.java
+----
+include::example$spring-eventsourced-shopping-cart/src/main/java/com/example/shoppingcart/ShoppingCartEntity.java[tag=class]
+----
+<1> Create a class that extends `EventSourcedEntity<S>`, where `S` is the state type this entity will store (i.e. `ShoppingCart`).
+<2> Make sure to annotate such class with `@Entity` and pass in an `entityKey` and an `entityType`.
+<3> Make use of Spring's RequestMapping annotation to define the routes to your entity.
+
+
+NOTE: The entity key `cartId` must match a path parameter (i.e. `cartId`) and such value needs to be unique per entity. On the other hand, the entity type `shopping-cart` is common for all instances of this entity but must be stable - cannot be changed after a production deploy - and unique across the different entity types.
+
+
+
+=== Updating state
+
+Having created the basis of our entity, we will now define how each command is handled. In the example below, we define a new endpoint that will add a new line item to a given shopping cart. It returns an `Effect` to emit an event and then sends a reply once the event is stored successfully. The state is updated by the event handler.
+
+IMPORTANT: The **only** way for a command handler to modify the Entity's state is by emitting an event. Any modifications made directly to the state (or instance variables) from the command handler are not persisted. When the Entity is passivated and reloaded, those modifications will not be present.
+
+[source,java,indent=0]
+.src/main/java/com/example/shoppingcart/ShoppingCartEntity.java
+----
+include::example$spring-eventsourced-shopping-cart/src/main/java/com/example/shoppingcart/ShoppingCartEntity.java[tag=addItem]
+----
+<1> The validation ensures the quantity of items added is greater than zero and it fails for calls with illegal values by returning an `Effect` with `effects().error`.
+<2> From the current incoming `LineItem` we create a new `ItemAdded` event representing the change of the cart.
+<3> We store the event by returning an `Effect` with `effects().emitEvent`.
+<4> The acknowledgment that the command was successfully processed is only sent if the event was successfully stored and processed, otherwise there will be an error reply. The lambda parameter `newState` gives us access to the new state returned by the processing of such event.
```

Comment:
Updated this to address your [comment](https://github.com/lightbend/kalix-jvm-sdk/pull/1201/files#r1004239371) @octonato . Does this look fine to you?

Reply:
That's better, but I think we should say applying the event instead of processing. We apply the event on the state and produce a new state. That's how we usually explain it. I'm sure we say that in other parts of the documentation as well.